docs/recipes/cache.md

```diff
@@ -8,37 +8,125 @@ service.
 }
 
-@State<ZooStateModel>({
-  defaults: {
-    zebras: []
+@State<Novel[]>({
+  name: 'novels',
+  defaults: []
+})
+export class NovelsState {
+
+  constructor(private novelsService: NovelsService) {}
+
+  @Action(GetNovels)
+  getNovels(ctx: StateContext<Novel[]>) {
+    return this.novelsService.getNovels().pipe(
+      tap(novels => ctx.setState(novels))
+    );
   }
+
+}
+```
+
+Imagine that this state of novels contains only minimal information about them such as ID and name.
+When the user selects a particular novel - he is redirected to a page with full information about this novel.
+We want to load this information only once. Let's create a state and call it `novelsInfo`, this will be
+the object whose keys are the identifiers of the novels:
+
+```ts
+import { State, Action, StateContext, createSelector } from '@ngxs/store';
+import { tap } from 'rxjs/operators';
+
+export interface NovelsInfoStateModel {
+  [key: string]: Novel;
+}
+
+export class GetNovelById {
+  static readonly type = '[Novels info] Get novel by ID';
+  constructor(public id: string) {}
+}
+
+@State<NovelsInfoStateModel>({
+  name: 'novelsInfo',
+  defaults: {}
 })
-export class ZooState {
-
-  constructor(private animalService: AnimalService) {}
-
-  @Action(GetZebra)
-  getZebra(ctx: StateContext<ZooStateModel>, action: GetZebra) {
-    const state = ctx.getState();
-    // payload = id of animal
-    const idx = state.zebras.findIndex(zebra => zebra.id === action.id);
-    if (idx > -1) {
-      // if we have the cache, just return it from the store
-      return dispatch(new GetZebraSuccess(state.zebras[idx]));
-    } else {
-      return this.animalService.getZebra(action.id).pipe(
-        map(resp => dispatch(new GetZebraSuccess(resp)))
-      );
+export class NovelsInfoState {
+
+  static getNovelById(id: string) {
+    return createSelector(
+      [NovelsInfoState],
+      (state: NovelsInfoStateModel) => state[id]
+    );
+  }
+
+  constructor(private novelsService: NovelsService) {}
+
+  @Action(GetNovelById)
+  getNovelById(ctx: StateContext<NovelsInfoStateModel>, action: GetNovelById) {
+    const novels = ctx.getState();
+    const id = action.id;
+
+    if (novels[id]) {
+      // If the novel with ID has been already loaded
+      // we just break the execution
+      return;
     }
+
+    return this.novelsService.getNovelById(id).pipe(
+      tap(novel => {
+        ctx.patchState({ [id]: novel });
+      })
+    );
   }
 
 }
 ```
+
+The component, that displays information about the novel, can subscribe to the `params` observable of the `ActivatedRoute` to listen to the params change. The code will look as following:
+
+```ts
+@Component({
+  selector: 'app-novel',
+  template: `
+    <h1>{{ novel?.title }}</h1>
+    <span>{{ novel?.author }}</span>
+    <p>
+      {{ novel?.content }}
+      <del datetime="{{ novel?.publishedAt }}"></del>
+    </p>
+  `
+})
+export class NovelComponent implements OnDestroy {
```

Comment:
@splincode do you think it's a good example using ActivatedRoute.params + switchMap?

Reply:
why not?